pkg/assets/store.go

```diff
@@ -251,14 +221,16 @@ func (s *StoreBuilder) AddSafeAuthorizationCredentials(ctx context.Context, name
 		return err
 	}
 
-	err := s.addToken(ctx, namespace, auth.Credentials, key)
-	if err != nil {
-		return fmt.Errorf("failed to get authorization token of type %q: %w", auth.Type, err)
+	if auth.Credentials.Name != "" {
```

Comment:
I've kept this check to preserve the behavior of the defunct `addToken()` which exited early if the name value was empty. Eventually we may want to reject secret/configmap selectors with empty name & key but it's another discussion to have.

Reply:
It doesn't make sense and it might be prevented for some core resources. Our CRDs don't validate though. It would be a nice improvement, at least for v1alpha1 resources like ScrapeConfig (poke @mviswanathsai).